clients/cli/src/main/java/org/apache/gravitino/cli/commands/TagEntity.java

```diff
@@ -102,4 +110,33 @@ public void handle() {
 
     System.out.println(entity + " now tagged with " + all);
   }
+
+  private void checkTags(Table gTable) {
+    String[] associatedTags = gTable.supportsTags().listTags();
+    checkRedundantTags(associatedTags);
+  }
+
+  private void checkTags(Schema gSchema) {
+    String[] associatedTags = gSchema.supportsTags().listTags();
+    checkRedundantTags(associatedTags);
+  }
+
+  private void checkTags(Catalog gCatalog) {
+    String[] associatedTags = gCatalog.supportsTags().listTags();
+    checkRedundantTags(associatedTags);
+  }
```

Comment:
I wouldn't structure the code this way because this thin layer of `checkTags` can/should be compressed into the `handle` method.
We may want to split the `handle` method for catalog, schema and table, instead.
That method is growing and there are chunks of `if...else` in it.

By the way, I wouldn't call tags "redundant", you are checking if tags are "duplicated". This is a nit though.